### PR TITLE
Fixes #142: Adding "ignore-duplicates" option for dependency resolution.

### LINKED ISSUE #142: Introduce 'skip' resolution strategy
This is a feature request to introduce a new 'skip' resolution strategy that will cause the version defined in the root composer.json to win.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

**Comment by grasmash** [comment]
@bd808 Let me know what you think.

I've added a new test to cover the new `skip` option. This will resolve #142.

The composer-merge-plugin will now have three dependency resolution strategies available:

Default. Uses composer resolver.
Replace. Duplicate package declarations in merged files replace root declarations.
Skip. Duplicate package declarations in merged files are skipped.

File: README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ Usage
             ],
             "recurse": true,
             "replace": false,
+            "skip": false,
```
Comment (bd808) [inline README.md]: I think calling this option `ignore-duplicates` might be a bit more intuitive than `skip`. There should probably also be documentation that points out that this feature flag and `replace` flag are mutually exclusive. It actually might be a good idea to add a check when reading the configuration and at least log a warning if both `replace` and `ignore-duplicates` are provided and true.

I worry a bit that we are growing the set of feature flags to the point that it will be hard for anyone to reason about which flags to set for a given project, but this one additional flag is probably not the breaking point for the current system. A reasonable 2.0 breaking API change would be to combine the `replace` and `ignore-duplicates` flags into a new `merge-strategy` setting that would encompass the `resolve`, `replace`, and `ignore` behaviors.

File: README.md [hunk 2]
```diff
@@ -118,6 +119,13 @@ package declarations found in merged files will overwrite the declarations
 made by earlier files. Files are loaded in the order specified by the
 `include` setting with globbed files being processed in alphabetical order.
 
+### skip
+
+By default, Composer's conflict resolution engine is used to determine which
+version of a package should be installed when multiple files specify the same
+package. A `"skip": true` setting can be provided to change to a "skip" 
+resolution strategy. In this mode, duplicate package declarations found in 
+merged files will be ignored, and the root file will take precedence.
```
Comment (bd808) [inline README.md]: > ### skip
>
> By default, Composer's conflict resolution engine is used to determine which
version of a package should be installed when multiple files specify the same
package. A `"skip": true` setting can be provided to change to a "skip" 
resolution strategy. In this mode, duplicate package declarations found in 
merged files will be ignored, and the root file will take precedence.

Change to:

### ignore-duplicates

By default, Composer's conflict resolution engine is used to determine which version of a package should be installed when multiple files specify the same package. An `"ignore-duplicates": true` setting can be provided to change to a "first version specified wins" conflict resolution strategy. In this mode, duplicate package declarations found in merged files will be ignored in favor of the declarations made by earlier files. Files are loaded in the order specified by the `include` setting with globbed files being processed in alphabetical order.

Note: `"replace": true` and `"ignore-duplicates": true` modes are mutually exclusive. If both are set, `"ignore-duplicates": true` will be used.

File: tests/phpunit/MergePluginTest.php [hunk 3]
```diff
@@ -188,6 +188,44 @@ function ($args) use ($that) {
...
+        $root->getConflicts()->shouldNotBeCalled();
+        $root->getReplaces()->shouldNotBeCalled();
+        $root->getProvides()->shouldNotBeCalled();
+        $root->getSuggests()->shouldNotBeCalled();
+
+        $extraInstalls = $this->triggerPlugin($root->reveal(), $dir);
+
+        $this->assertEquals(0, count($extraInstalls));
+    }
+
```
Comment (bd808) [inline tests/phpunit/MergePluginTest.php]: Add a second test with both ignore-duplicates and replace enabled that ensures that ignore-duplicates wins.

**Comment by grasmash** [comment]
@bd808 Thanks for the feedback. All good suggestions, I'll make changes and push.

**Comment by grasmash** [comment]
@bd808 All feedback has been addressed!

**Comment by grasmash** [comment]
@bd808 I think this is good to go. GitHub only has a red "x" because I didn't change a particular line that you commented on in your review. But, I added a test after than line.
